src/pages/index.js

```diff
@@ -63,48 +100,158 @@ Array.from(document.forms).forEach((formElement) => {
     event.preventDefault();
-    userProfile.setUserInfo(name, description);
-    popupProfile.close();
+    popupProfileInfo.setRender(true); // renders faster when synced before the async begins
+    api
+      .setUserProfileData(name, description)
+      .then(() => {
+        userProfile.setUserInfo(name, description);
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        popupProfileInfo.close();
+        popupProfileInfo.setRender(false);
+      });
   },
-  configuration.inputSelector
+  configuration.inputSelector,
+  submitProfileInfo,
+  "Saving..."
 );
-popupProfile.setEventListeners();
+popupProfileInfo.setEventListeners();
+/* END PROFILE INFO SECTION */
 
-buttonEditProfile.addEventListener("click", () => {
-  const { name, description } = userProfile.getUserInfo();
-  inputProfileName.value = name;
-  inputProfileDescription.value = description;
-  formValidators["profile_form"].resetFormValidation(); // input always valid when taking from the html page
-  popupProfile.open();
+/* PROFILE AVATAR SECTION */
+buttonUpdateAvatar.addEventListener("click", () => {
+  inputProfileAvatar.value = profileAvatar.src;
+  popupProfileAvatar.open();
 });
+const popupProfileAvatar = new PopupWithForm(
+  modalWindowProfileAvatar,
+  (event, { avatar }) => {
+    event.preventDefault();
+    popupProfileAvatar.setRender(true);
+    api
+      .setUserProfileAvatar(avatar)
+      .then(() => {
+        profileAvatar.src = avatar;
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        popupProfileAvatar.close();
+        popupProfileAvatar.setRender(false);
+      });
+  },
+  configuration.inputSelector,
+  submitProfileAvatar,
+  "Saving..."
+);
+popupProfileAvatar.setEventListeners();
+/* END PROFILE AVATAR SECTION */
 
 /* CARD ADD SECTION */
+buttonAddCard.addEventListener("click", () => {
+  // don't show errors upon opening the modal when no input is issued
+  popupCardAdd.open();
+});
+
 const popupCardAdd = new PopupWithForm(
   modalWindowCardAdd,
-  (event, cardInputs) => {
+  (event, { title, link }) => {
     event.preventDefault();
-    const cardElement = createCard(cardInputs);
-    cardSection.addItem(cardElement, "prepend"); // can still be used to add later cards, not just init ones
-    popupCardAdd.close();
-    formCardAdd.reset();
-    formValidators["card-add_form"].disableButton();
+    popupCardAdd.setRender(true);
+    api
+      .addNewCard(title, link)
+      .then((newCard) => {
+        const cardElement = createCard(newCard);
+        cardSection.addItem(cardElement, "prepend"); // can still be used to add later cards, not just init ones
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        popupCardAdd.close();
+        popupCardAdd.setRender(false);
+        formCardAdd.reset();
+        formValidators["card-add_form"].disableButton();
```

Comment:
```js

        popupCardAdd.close();
        formCardAdd.reset();
        formValidators["card-add_form"].disableButton();
```

These 3 lines of code should be in line 189 in a `then` block

You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` or in `finally` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

-----------------

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.